### PR TITLE
fix(compiler): add math elements to schema

### LINKED ISSUE #55608: MathML is unsupported
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

When trying to use [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) in an angular template you get a warning that `<math>` it's an unrecognised element. If you disable this warning by adding `schemas: [NO_ERRORS_SCHEMA]` it doesn't render correctly. At first glance they seem correct because the produced elements are in the MathML namespace (something other frameworks can get wrong) but the element's prototype is just `Element` rather than `MathMLElement`.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/github-tdmxzz?file=src%2Fmain.ts

### Please provide the exception or error you saw

```true
Error in src/main.ts (12:3)
':math:math' is not a known element:
1. If ':math:math' is an Angular component, then verify that it is included in the '@Component.imports' of this component.
2. To allow any element add 'NO_ERRORS_SCHEMA' to the '@Component.schemas' of this component.
```
```


### Please provide the environment you discovered this bug in (run `ng version`)

_No response_

### Anything else?

The template I tried is below. 

```html
<math>
  <mfrac>
    <mn>1</mn>
    <msqrt>
      <mn>2</mn>
    </msqrt>
  </mfrac>
</math>
```

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes that we didn't have the MathML elements in the schema. Note that we can't discover which tag names are available by looking at globally-available classes, because all MathML elements are `MathMLElement` rather than something like `SVGCircleElement`. As such, I ended up having to hardcode the currently-available tags.

Fixes #55608.

### CODE REVIEW

File: packages/compiler/test/schema/schema_extractor.ts [hunk 1]
```diff
@@ -154,6 +159,17 @@ export function extractSchema(): Map<string, string[]> | null {
     descMap.set(elHierarchy, MISSING_FROM_CHROME[elHierarchy]);
   });
 
+  // Math top level
+  const math = document.createElementNS('http://www.w3.org/1998/Math/MathML', 'math');
+  extractProperties(MathMLElement, math, visited, descMap, MATH_PREFIX, HTMLELEMENT_IF);
+
+  // This script is written under the assumption that each tag has a correspond class name, e.g.
```
Comment (JoostK) [inline packages/compiler/test/schema/schema_extractor.ts:166]:
```suggestion
  // This script is written under the assumption that each tag has a corresponding class name, e.g.
```

File: packages/compiler/test/schema/schema_extractor.ts [hunk 2]
```diff
@@ -166,7 +182,10 @@ function assertNoMissingTags(descMap: Map<string, string[]>): void {
     extractedTags.push(...key.split('|')[0].split('^')[0].split(','));
   });
 
-  const missingTags = ALL_HTML_TAGS.split(',').filter((tag) => extractedTags.indexOf(tag) == -1);
+  const missingTags = [
+    ...ALL_HTML_TAGS.split(','),
+    ...ALL_MATH_TAGS.split(',').map((tag) => MATH_PREFIX + tag),
+  ].filter((tag) => extractedTags.indexOf(tag) == -1);
```
Comment (JoostK) [inline packages/compiler/test/schema/schema_extractor.ts:188]: If you feel like changing this

```suggestion
  ].filter((tag) => !extractedTags.includes(tag));
```

**Review by JoostK — APPROVED** [review]

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: fw-security

**Review by josephperrott — APPROVED** [review]
LGTM

Reviewed-for: fw-security

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 2e891ad72a6dd4f7f0707a51990cff63ec02f4ce.

**Pull request closed without merging**